src/state.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright: ConsensusLab
+//
+use cid::Cid;
+use fil_actors_runtime::{ActorDowncast, Map, SYSTEM_ACTOR_ADDR};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::Cbor;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::address::Address;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::MethodNum;
+use ipc_gateway::{IPCAddress, StorableMsg, SubnetID};
+use num_traits::Zero;
+use primitives::{TCid, THamt, TLink};
+use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
+use std::str::FromStr;
+
+use crate::exec::{AtomicExec, AtomicExecParams, AtomicExecParamsMeta};
+use crate::types::CrossMsgParams;
+use crate::{atomic, ConstructorParams};
+
+#[derive(Serialize, Deserialize)]
+pub struct State {
+    pub network_name: SubnetID,
+    pub atomic_exec_registry: TCid<THamt<Cid, AtomicExec>>,
+    pub nonce: u64,
+    pub ipc_gateway_address: Address,
+    pub ipc_gateway_cross_method_num: MethodNum,
```

Comment:
This is the `MethodNum` identifying the the `CrossSend` from the `ipc_gateway`, right? We can probably make this a constant (I don't expect to change between gateway deployments).

Reply:
Yeah, we can just use `ipc_gateway::Method::SendCross`.